bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -848,6 +850,21 @@ protected void runRule(RuntimeRule rule, RuleEngineCallbackImpl.TriggerData td)
 
     }
 
+    private void clearCacheOutputValues(RuntimeRule rule) {
+        List<Trigger> tiggers = rule.getTriggers();
+        for (Trigger trigger : tiggers) {
+            ((RuntimeTrigger) trigger).setOutputs(null);
```

Comment:
The triggers and actions outputs have to be cleared before new rule execution, because the setTriggerOutpus() method sets outputs on the current triggered trigger, but outputs form previous execution have to be cleaned.